docs/src/pages/api-reference/react.mdx

```diff
@@ -89,11 +95,20 @@ components. It takes your File Router as a generic
 import { generateReactHelpers } from "@uploadthing/react/hooks";
 
 import type { OurFileRouter } from "~/app/api/uploadthing/core";
+import { getBaseUrl } from "~/app/utils/getBaseUrl";
 
 export const { useUploadThing, uploadFiles } =
-  generateReactHelpers<OurFileRouter>();
+  generateReactHelpers<OurFileRouter>({
+    url: getBaseUrl() + "/api/uploadthing",
+  });
 ```
 
+### Configuration
+
+| Prop | Type   | Required | Notes           | Description                                               |
+| :--- | :----- | :------- | :-------------- | :-------------------------------------------------------- |
+| url  | string | Yes      | Added in `v6.0` | The **absolute** URL to the API handler you created above |
```

Comment:
I highly dislike making this required, will break all current users for no good reason. Also we can presume 95%+ of users will be on the local path of `/api/uploadthing`

Reply:
So you think we should have a default like

```ts
const defaultUrl = process.env.VERCEL_URL 
  ? `https://$${process.env.VERCEL_URL}/api/uploadthing`
  : `http://localhost:${process.env.PORT ?? 3000}/api/uploadthing`
```

Problem is then we use `process.env` on client which isn't supported in Vite based apps, so we'd need a workaround to handle that too.

@markflorkowski was fine doing a major for this, so a breaking change isn't too bad, especially since there's good migration docs. I also think it's reasonable to think people can provide this themselves, especially since we're removing the option on the backend where it can be automatically detected (think trpc links where url is required and you can deploy your trpc router anywhere, even if the majority will choose /api/[trpc])

Reply:
> will break all current users

Major version bumps are allowed to do that IMO.

> Also we can presume 95%+ of users will be on the local path of /api/uploadthing

But we need the absolute URL, and there is not a consistent guaranteed way of constructing that in arbitrary environments. 


Perhaps we could fall back to old behaviour and throw terrible errors in the console if you try to just bump the version without adding the required config?